### PR TITLE
Issue 6 revitalize component

### PR DESCRIPTION
**Pull request by shulkaolka**
#6

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -34,6 +34,18 @@ class Client {
     return await this.s3.copyObject(params).promise();
```
Comment (kirill-levitskiy) [inline lib/client.js]: Outside of try/catch blocks, `return await` is redundant. There's even an [ESLint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) to detect it, but it allows it in try/catch.

File: lib/client.js [hunk 2]
```diff
@@ -34,6 +34,18 @@ class Client {
...
   }
 
+  async listObjects(params) {
+    return await this.s3.listObjects(params).promise();
```
Comment (kirill-levitskiy) [inline lib/client.js:38]: Outside of try/catch blocks, `return await` is redundant. There's even an [ESLint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) to detect it, but it allows it in try/catch.

File: lib/client.js [hunk 3]
```diff
@@ -34,6 +34,18 @@ class Client {
...
+  }
+
+  async getObject(bucketName, fileName) {
+    const params = {
+      Bucket: bucketName,
+      Key: fileName,
+    };
+    return await this.s3.getObject(params).promise();
```
Comment (kirill-levitskiy) [inline lib/client.js]: Outside of try/catch blocks, `return await` is redundant. There's even an [ESLint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) to detect it, but it allows it in try/catch.

File: lib/client.js [hunk 4]
```diff
@@ -44,7 +56,7 @@ class Client {
       Key: fileName,
       Body: body,
     };
-    await this.s3.upload(params).promise();
+    return await this.s3.upload(params).promise();
```
Comment (kirill-levitskiy) [inline lib/client.js]: Outside of try/catch blocks, `return await` is redundant. There's even an [ESLint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) to detect it, but it allows it in try/catch.

File: lib/actions/streamToFile.js [hunk 5]
```diff
@@ -47,15 +40,7 @@ exports.process = async function (msg, cfg) {
...
-    Body: passthrough,
-  };
-
-  debug('inserting into s3: %j', s3Input);
-
-  const result = await s3.upload(s3Input).promise();
-
+  const fileName = msg.body.filename ? msg.body.filename : attachments[0].key.replace(/\//g, '');
+  const result = await client.upload(bucketName, fileName, passthrough);
   this.emit('data', messages.newMessageWithBody(result));
```
Comment (kirill-levitskiy) [inline lib/actions/streamToFile.js]: Do we need to use `await` before `this.emit`?

**Review by kirill-levitskiy — CHANGES_REQUESTED** [review]

File: component.json [hunk 6]
```diff
@@ -30,9 +30,10 @@
       "fields": {
         "bucketName": {
           "viewClass": "TextFieldView",
-          "label": "Default Bucket Name",
+          "label": "Default Bucket Name and folder",
```
Comment (A3a3e1) [inline component.json:33]:
```suggestion
          "label": "Default Bucket Name and Folder",
```

**Review by A3a3e1 — APPROVED** [review]

**Review by kirill-levitskiy — APPROVED** [review]

**Pull request merged**